test/integration/firewall/firewallBase.test.ts

```diff
@@ -0,0 +1,623 @@
+    const elsaWallet = testContext.elsa
+    const elsaHex = AccountID.from(elsaWallet.classicAddress).toHex()
+    const currencyHex = Currency.from('USD').toHex()
+    const hook1Param1 = new iHookParamEntry(
+      new iHookParamName('FWP'),
+      new iHookParamValue(elsaHex, true)
+    )
+    const hook1Param2 = new iHookParamEntry(
+      new iHookParamName('FIC'),
+      new iHookParamValue(currencyHex, true)
+    )
+    const hook1Param3 = new iHookParamEntry(
+      new iHookParamName('FIP'),
+      new iHookParamValue(elsaHex, true)
+    )
+    const hook1Param4 = new iHookParamEntry(
+      new iHookParamName('FI'),
+      new iHookParamValue(calculateHookOn(['Payment']), true)
+    )
+    const hook1Param5 = new iHookParamEntry(
+      new iHookParamName('FLD'),
+      new iHookParamValue(floatToLEXfl('1'), true)
+    )
+    const hook1Param6 = new iHookParamEntry(
+      new iHookParamName('FUD'),
+      new iHookParamValue(floatToLEXfl('100'), true)
+    )
+    const hook1 = createHookPayload({
+      version: 0,
+      createFile: 'firewall_base',
+      namespace: 'firewall_base',
+      flags: SetHookFlags.hsfOverride,
+      hookOnArray: ['Payment'],
+      hookParams: [
+        hook1Param1.toXrpl(),
+        hook1Param2.toXrpl(),
+        hook1Param3.toXrpl(),
+        hook1Param4.toXrpl(),
+        hook1Param5.toXrpl(),
+        hook1Param6.toXrpl(),
+      ],
+    })
+    await setHooksV3({
+      client: testContext.client,
+      seed: testContext.alice.seed,
+      hooks: [{ Hook: hook1 }],
+    } as SetHookParams)
+    const hook2 = createHookPayload({
+      version: 0,
+      createFile: 'firewall_provider',
+      namespace: 'firewall_provider',
+      flags: SetHookFlags.hsfOverride,
+      hookOnArray: ['Invoke'],
+    })
+    hook2.HookNamespace =
+      '0000000000000000000000000000000000000000000000000000000000000000'
+    await setHooksV3({
+      client: testContext.client,
+      seed: testContext.elsa.seed,
+      hooks: [{ Hook: hook2 }],
+    } as SetHookParams)
+
+    // INVOKE OUT
+    const tx1Param1 = new iHookParamEntry(
+      new iHookParamName('M'),
+      new iHookParamValue(floatToLEXfl('1'), true)
+    )
+    const tx1blob = formatAccountBlob([bobWallet.classicAddress])
+    const builtTx1: Invoke = {
+      TransactionType: 'Invoke',
+      Account: elsaWallet.classicAddress,
+      Blob: tx1blob,
+      HookParameters: [tx1Param1.toXrpl()],
+    }
+    const result1 = await Xrpld.submit(testContext.client, {
+      wallet: elsaWallet,
+      tx: builtTx1,
+    })
+
+    const hook1Emitted = await ExecutionUtility.getHookExecutionsFromMeta(
+      testContext.client,
+      result1.meta as TransactionMetadata
+    )
+    expect(hook1Emitted.executions[0].HookReturnString).toMatch('')
+
+    // INVOKE OUT
+    const tx2Param1 = new iHookParamEntry(
+      new iHookParamName('M'),
+      new iHookParamValue(floatToLEXfl('2'), true)
+    )
+    const tx2blob = formatAccountCurrencyBlob('USD', [
```

Comment:
formatAccountCurrencyBlob does not accept string I think this needs to be an array `['USD']`

Reply:
Thats actually for a different version of the firewall, one where I added the ability to force a specific Issued Currency. We can add that as v2 but for now I have to stick to the original spec.